packages/openapi-v3/src/__tests__/unit/json-to-schema.unit.ts

```diff
@@ -139,6 +139,31 @@ describe('jsonToSchemaObject', () => {
     expect(jsonToSchemaObject(inputDef)).to.eql(expectedDef);
   });
 
+  it('handles circular references', () => {
+    const schemaJson: JsonSchema = {
+      title: 'ReportState',
+      properties: {
+        states: {type: 'array', items: {$ref: '#/definitions/ReportState'}},
+        benchmarkId: {type: 'string'},
+        color: {type: 'string'},
+      },
+    };
+    schemaJson.definitions = {ReportState: schemaJson};
```

Comment:
Is it necessary for `schemaJson.definitions` to contain an entry for `ReportState`? What will happen when there is no such entry? Can you add a test for that please?

Reply:
I improved the tests with different combinations.